### PR TITLE
docs: add WITH keyword

### PR DESCRIPTION
**Pull request by TheTanc**

### CODE REVIEW

File: docs/reference/configuration/server.md [hunk 1]
```diff
@@ -15,7 +15,7 @@ configuration file as follows:
 ```shell
 aaa.bbb.ccc.ddd=myValue
 ```
-
+``
```
Comment (mpsq) [inline docs/reference/configuration/server.md:18]: is that intended?

File: sidebars.js [hunk 2]
```diff
@@ -135,6 +135,7 @@ module.exports = {
             "reference/sql/timestamp",
             "reference/sql/truncate",
             "reference/sql/union",
+            "reference/sql/with",
             "reference/sql/where",
```
Comment (mpsq) [inline sidebars.js:139]: with after where

File: docs/reference/sql/with.md [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+sub-queries are used several times. 
+
+## Syntax
+
+ ![Flow chart showing the syntax of the WITH clause](/img/docs/diagrams/with.svg)
+ 
+ Where:
+ - `subQueryName` is the alias for the sub-query
+ - `subQuery` is a SQL query (e.g `SELECT * FROM table`)
+ - `mainQuery` is the main SQL query which involves the `subQuery` using its alias.
```
Comment (mpsq) [inline docs/reference/sql/with.md:19]:
```suggestion
 - `mainQuery` is the main SQL query which involves the `subQuery` using its alias
```

File: docs/reference/configuration/server.md (hunk 1 above)
Comment (TheTanc) [reply docs/reference/configuration/server.md]: nope

**Pull request merged**